runtime/caches/redis.ts

```diff
@@ -1,36 +1,28 @@
-import { Redis } from "https://deno.land/x/upstash_redis@v1.22.1/mod.ts";
-
-import { logger, tracer } from "../../observability/otel/config.ts";
 import {
   assertCanBeCached,
   assertNoOptions,
-  withCacheNamespace,
+  requestURLSHA1,
 } from "./common.ts";
+import { createClient } from "npm:redis@^4.5";
 
-const redisUrl = Deno.env.get("UPSTASH_REDIS_REST_URL");
-const redisToken = Deno.env.get("UPSTASH_REDIS_REST_TOKEN");
-
-export const redis = redisUrl && redisToken
-  ? new Redis({
-    url: redisUrl,
-    token: redisToken,
-    enableTelemetry: true,
-  })
-  : null;
-
-interface ResponseMetadata {
-  body: string;
-  status: number;
-  headers: [string, string][];
-}
+const REDIS_URL = Deno.env.get("REDIS_URL") || "redis://localhost:6379/0";
```

Comment:
mistyped or expected?
```suggestion
const REDIS_URL = Deno.env.get("REDIS_URL") || "redis://localhost:6379/";
```

Reply:
I know the default database will always be 0, but I want to make sure it's explicit for everyone.